src/containers/edit-profile/profile-tab/profile-tabs/profile-general-tab/ProfileGeneralTab.tsx

```diff
@@ -38,47 +55,101 @@ const ProfileGeneralTab: FC<ProfileGeneralTabProps> = ({ user }) => {
     data,
     errors
   } = useForm<EditProfileForm>({
-    initialValues: getInitialValues(user),
+    initialValues,
     validations
   })
 
+  const updatedData: UpdateUserParams = useMemo(
+    () => ({
+      firstName: data.firstName,
+      lastName: data.lastName,
+      address: {
+        country: data.country ?? '',
+        city: data.city ?? ''
+      },
+      professionalSummary: data.professionalSummary,
+      mainSubjects:
+        user.role[0] !== UserRoleEnum.Admin
+          ? user.mainSubjects[user.role[0]]
+          : [],
+      nativeLanguage: data.nativeLanguage ?? null,
+      videoLink: data.videoLink
+    }),
+    [data, user.mainSubjects, user.role]
+  )
+
+  useEffect(() => {
+    handleProfileData(updatedData, errors)
+  }, [updatedData, errors, handleProfileData])
+
   const onLanguageChange = (
     _: SyntheticEvent,
     value: EditProfileForm['nativeLanguage']
   ) => {
     handleNonInputValueChange('nativeLanguage', value)
   }
 
-  return (
-    <Box sx={styles.root}>
-      <TitleWithDescription
-        description={t('editProfilePage.profile.description')}
-        style={styles.headerTitleWithDesc}
-        title={t('editProfilePage.profile.title')}
-      />
+  const resizeImage = (photo: File) => {
+    const originalPhotoPath = URL.createObjectURL(photo)
+    const photoSizes = { newWidth: 440, newHeight: 440 }
+
+    imageResize(originalPhotoPath, photoSizes)
+      .then((resizedPhoto) => {
+        handleProfileData({
+          ...updatedData,
+          photo: { src: resizedPhoto, name: photo.name }
+        })
+      })
+      .catch(console.error)
+  }
+
+  const addPhoto = ({ files, error }: Emitter) => {
+    if (error) {
+      setAlert({
+        severity: snackbarVariants.error,
+        message: error
+      })
+      return
+    }
+
+    resizeImage(files[0])
+  }
+
+  const handleRemovePhoto = () => {
+    handleProfileData({
+      ...updatedData,
+      photo: null
+    })
+  }
 
+  const photoToDisplay =
+    photo === null
+      ? ''
+      : photo?.src ||
+        (user.photo && `${import.meta.env.VITE_APP_IMG_USER_URL}${user.photo}`)
```

Comment:
Maybe use `createUrlPath` helper function here instead?

Reply:
Unfortunately, it doesn't suit here. 
Current result looks something like: `path/photo`. Using the function, I get an extra slash: `path//photo`.
It's because the `VITE_APP_IMG_USER_URL` variable contains a slash at the end. But I believe it's better not to change the variable or at least to do it in a separate PR, as it requires changes in many files.

Reply:
Our helper function is a bit strange. I would refactor it to concat urls correctly, for example using `new URL(path, base).href`, which concats urls correctly regardless of slash at the end. Possible candidate for a new issue? :)